### PR TITLE
7-CORE-POST-api-articles-article_id-comments

### PR DESCRIPTION
**Pull request by miors**
endpoint /api/articles/:article_id/comments
added solution, test
updated endpoints doc

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -164,3 +164,57 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    };
+
+    return request(app)
+      .post("/api/articles/9/comments")
+      .send(postJson)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(typeof comment.comment_id).toBe("number");
+        expect(typeof comment.body).toBe("string");
+        expect(typeof comment.article_id).toBe("number");
```
Comment (pippaamy) [inline __tests__/integration.test.js:182]: Great job - we also know the article id is 9 :)

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -164,3 +164,57 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+    };
+
+    return request(app)
+      .post("/api/articles/999999/comments")
+      .send(postJson)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("ID not presence in DB");
+      });
+  });
```
Comment (pippaamy) [inline __tests__/integration.test.js:204]: lovely error testing!

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -164,3 +164,57 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      username: "butter_bridge",
+      body: "Hello, this is a comment",
+    };
+
+    return request(app)
+      .post("/api/articles/invalidInput/comments")
+      .send(postJson)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid input");
```
Comment (pippaamy) [inline __tests__/integration.test.js:217]: what if the body of the message is missing? 

what if the username doesnt exist??

can we pop in some tests for these :)

File: endpoints.json [hunk 4]
```diff
@@ -59,5 +59,25 @@
...
+  "POST /api/articles/9/comments": {
+    "description": "add a comment for an article",
+    "queries": [],
+    "request_body": {
+      "username": "butter_bridge",
+      "body": "Hello, this is a comment"
+    },
+    "exampleResponse": {
+      "comments": [
+        {
```
Comment (pippaamy) [inline endpoints.json:72]: well remembered !

**Review by pippaamy — COMMENTED** [review]
Great job, I would add those two other error tests :)

File: __tests__/integration.test.js (hunk 1 above)
Comment (miors) [reply __tests__/integration.test.js:182]: Added additional expect for article id is 9 :)

**Pull request merged**